mongorestore/mongorestore_test.go

```diff
@@ -250,6 +251,110 @@ func TestMongorestore(t *testing.T) {
 	})
 }
 
+func TestMongoRestoreSpecialCharactersNameSpaces(t *testing.T) {
+	testtype.SkipUnlessTestType(t, testtype.IntegrationTestType)
+
+	session, err := testutil.GetBareSession()
+	if err != nil {
+		t.Fatalf("No server available")
+	}
+
+	Convey("With a test MongoRestore", t, func() {
+		args := []string{
+			NumParallelCollectionsOption, "1",
+			NumInsertionWorkersOption, "1",
+		}
+
+		restore, err := getRestoreWithArgs(args...)
+		So(err, ShouldBeNil)
+
+		db := session.Database("db1")
+		Convey("and majority is used as the default write concern", func() {
+			So(db.WriteConcern(), ShouldResemble, writeconcern.New(writeconcern.WMajority()))
+		})
+
+		longCollection := db.Collection(specialCharactersCollectionName)
+		longCollection.Drop(nil)
+
+		Convey("and --nsInclude a collection name with special characters", func() {
+			restore.TargetDirectory = "testdata/specialcharacter"
+			restore.NSOptions.NSInclude = make([]string, 1)
+			restore.NSOptions.NSInclude[0] = "db1." + specialCharactersCollectionName
+
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+			So(result.Successes, ShouldEqual, 1)
+			So(result.Failures, ShouldEqual, 0)
+
+			count, err := longCollection.CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 1)
+
+		})
+
+		Convey("and --nsExclude a collection name with special characters", func() {
+			restore.TargetDirectory = "testdata/specialcharacter"
+			restore.NSOptions.NSExclude = make([]string, 1)
+			restore.NSOptions.NSExclude[0] = "db1." + specialCharactersCollectionName
+			result := restore.Restore()
+			So(result.Err, ShouldBeNil)
+			So(result.Successes, ShouldEqual, 0)
+			So(result.Failures, ShouldEqual, 0)
+
+			count, err := longCollection.CountDocuments(nil, bson.M{})
+			So(err, ShouldBeNil)
+			So(count, ShouldEqual, 0)
+
+		})
+
+		Convey("and --nsTo a collection name without special characters "+
+			"--nsFrom a collection name with special characters", func() {
+			restore.TargetDirectory = "testdata/specialcharacter"
+			restore.NSOptions.NSFrom = make([]string, 1)
+			restore.NSOptions.NSFrom[0] = "db1." + specialCharactersCollectionName
+			restore.NSOptions.NSTo = make([]string, 1)
+			restore.NSOptions.NSTo[0] = "db1.aCollectionNameWithoutSpecialCharacters"
+
+			shortCollection := db.Collection("aCollectionNameWithoutSpecialCharacters")
```

Comment:
```suggestion
			nonspecialCollection := db.Collection("aCollectionNameWithoutSpecialCharacters")
```
[nit] this also seems to be a copy-pasted name; not sure what to call it 😮 ; maybe `nonspecial` or `standard` or something.

Reply:
I seem to have missed a few more of these copy paste errors -- thanks for pointing it out!